kernel/src/syscall.rs

```diff
@@ -828,10 +833,11 @@ pub fn handle_inner(pid: PID, tid: TID, in_irq: bool, call: SysCall) -> SysCallR
 
                 if !phys_ptr.is_null() {
                     if mm.is_main_memory(phys_ptr) {
+                        let range_start = range.as_mut_ptr() as *mut usize;
+                        let range_end =
+                            range_start.wrapping_add(range.len() / core::mem::size_of::<usize>());
                         unsafe {
-                            range
```

Comment:
Adding a note on the significance of this change: this code gets a pointer to `u8`, which has its length divided by the size of `usize`, so the memory is not being fully-zeroed before being returned: only the bottom quarter is zeroed.